### ISSUE TITLE
Hot module reloading doesn't work consistently

### ISSUE THREAD

**Issue by vinceau** [vinceau/project-clippi#110]
I noticed that hot module reloading (HMR) doesn't work all the time anymore. I've managed to narrow it down to one of [these commits](https://github.com/vinceau/project-clippi/compare/053d2fb...2e4d3e0).

My understanding is that this issue with HMR likely stems from the fact that we are using a hacky workaround:

```js
if (isDevelopment) {
  app.prependOnceListener("ready", startUp);
} else {
  app.on("ready", startUp);
}
```

As a recap, in #100 we upgraded electron from 4.1.1 to 7.3.2, due to vulnerabilities found in Electron 4. The `twitch-electron-auth-provider` package works with peer dependency up to `< 9.0.0`, so we could have installed any Electron up til version 8. So why did I settle on 7.3.2? As evident in the commits, initially we had upgraded to Electron to 8.4.1, but it caused the console to be spammed with 'DevTools failed to parse SourceMap' warning messages. At the time of writing, it seems like there is still no fix available for this, and in addition, this issue also plagues later versions as well including version 9 and version 10.

And so, we're stuck with electron version 7 of Electron. But then Electron version 7 has this very strange bug with Windows 10 dark mode, where the ready event doesn't fire in dev mode (yes, really) and thus not opening the app. So we use this workaround [as described here](https://github.com/electron/electron/issues/19468#issuecomment-623529556) to get around opening the app in dev mode.

Which leads me to repeat that I believe this workaround is the cause of HMR not working. I'm still not sure what the best path forward in regards to this.

In conclusion: I hate Electron.

**Comment by vinceau** [comment]
Closing due to stale.
